ignite/metrics/metric.py

```diff
@@ -377,30 +378,33 @@ def iteration_completed(self, engine: Engine) -> None:
         """
 
         output = self._output_transform(engine.state.output)
-        if isinstance(output, Mapping):
-            if self.required_output_keys is None:
-                raise TypeError(
-                    f"Transformed engine output for {self.__class__.__name__} metric should be a tuple/list, "
-                    f"but given {type(output)}"
-                )
-            if not all([k in output for k in self.required_output_keys]):
-                raise ValueError(
-                    "When transformed engine's output is a mapping, "
-                    f"it should contain {self.required_output_keys} keys, but given {list(output.keys())}"
-                )
-            output = tuple(output[k] for k in self.required_output_keys)
-
-        if isinstance(output, Sequence) and all([_is_list_of_tensors_or_numbers(o) for o in output]):
-            if not (len(output) == 2 and len(output[0]) == len(output[1])):
-                raise ValueError(
-                    f"Output should have 2 items of the same length, "
-                    f"got {len(output)} and {len(output[0])}, {len(output[1])}"
-                )
-            for o1, o2 in zip(output[0], output[1]):
-                # o1 and o2 are list of tensors or numbers
-                tensor_o1 = _to_batched_tensor(o1)
-                tensor_o2 = _to_batched_tensor(o2, device=tensor_o1.device)
-                self.update((tensor_o1, tensor_o2))
+        if self._skip_unrolling is False:
+            if isinstance(output, Mapping):
+                if self.required_output_keys is None:
+                    raise TypeError(
+                        f"Transformed engine output for {self.__class__.__name__} metric should be a tuple/list, "
+                        f"but given {type(output)}"
+                    )
+                if not all([k in output for k in self.required_output_keys]):
+                    raise ValueError(
+                        "When transformed engine's output is a mapping, "
+                        f"it should contain {self.required_output_keys} keys, but given {list(output.keys())}"
+                    )
+                output = tuple(output[k] for k in self.required_output_keys)
+
+            if isinstance(output, Sequence) and all([_is_list_of_tensors_or_numbers(o) for o in output]):
```

Comment:
`self._skip_unrolling` should skip this if-clause but keep `if isinstance(output, Mapping):`

Reply:
Okay. You mean if skip unrolling is True, then only ``` if isinstance(output, Mapping):  ``` clause should run. If it is false then the ``` if isinstance(output, Sequence) ``` clause should run ?

Reply:
```       if isinstance(output, Mapping):
                if self.required_output_keys is None:
                    raise TypeError(
                        f"Transformed engine output for {self.__class__.__name__} metric should be a tuple/list, "
                        f"but given {type(output)}"
                    )
                if not all([k in output for k in self.required_output_keys]):
                    raise ValueError(
                        "When transformed engine's output is a mapping, "
                        f"it should contain {self.required_output_keys} keys, but given {list(output.keys())}"
                    )
                output = tuple(output[k] for k in self.required_output_keys)

        if self._skip_unrolling is False:
            if isinstance(output, Sequence) and all([_is_list_of_tensors_or_numbers(o) for o in output]):
                if not (len(output) == 2 and len(output[0]) == len(output[1])):
                    raise ValueError(
                        f"Output should have 2 items of the same length, "
                        f"got {len(output)} and {len(output[0])}, {len(output[1])}"
                    )
                for o1, o2 in zip(output[0], output[1]):
                    # o1 and o2 are list of tensors or numbers
                    tensor_o1 = _to_batched_tensor(o1)
                    tensor_o2 = _to_batched_tensor(o2, device=tensor_o1.device)
                    self.update((tensor_o1, tensor_o2))
            else:
                self.update(output)
        else:
            self.update(output) ```
            Sort of something like this?

Reply:
You join together two if-clauses:
```
if self._skip_unrolling is False:
        if isinstance(output, Sequence) and all([_is_list_of_tensors_or_numbers(o) for o in output]):
```

Reply:
```
    if (not self._skip_unrolling) and isinstance(output, Sequence) and all([_is_list_of_tensors_or_numbers(o) for o in output])  :
            if not (len(output) == 2 and len(output[0]) == len(output[1])):
                raise ValueError(
                    f"Output should have 2 items of the same length, "
                    f"got {len(output)} and {len(output[0])}, {len(output[1])}"
                )
            for o1, o2 in zip(output[0], output[1]):
                # o1 and o2 are list of tensors or numbers
                tensor_o1 = _to_batched_tensor(o1)
                tensor_o2 = _to_batched_tensor(o2, device=tensor_o1.device)
                self.update((tensor_o1, tensor_o2))

    else:
        self.update(output) 
 ```
 
Is it better now?

Reply:
yes, it is better now. Once you updated the code, please do not forget to reformat the code using ufmt: https://github.com/pytorch/ignite/blob/master/CONTRIBUTING.md#formatting-code to have compliant code